jolt-core/src/zkvm/dag/state_manager.rs

```diff
@@ -276,6 +276,16 @@ where
         }
     }
 
+    pub fn get_trace_len(&self) -> usize {
+        if let Some(ref verifier_state) = self.verifier_state {
+            verifier_state.trace_length
+        } else if let Some(ref prover_state) = self.prover_state {
+            prover_state.trace.len()
+        } else {
+            panic!("Neither prover nor verifier state initialized");
+        }
+    }
```

Comment:
```suggestion
pub fn get_trace_len(&self) -> usize {
    if let Some(verifier_state) = self.verifier_state.as_ref() {
        verifier_state.trace_length
    } else if let Some(prover_state) = self.prover_state.as_ref() {
        prover_state.trace.len()
    } else {
        panic!("Neither prover nor verifier state initialized");
    }
}
```

nit: does this work? I don't think we use `ref` anywhere else in code and I think in general `as_ref()` these days is much more common and possibly clearer. Same may apply else where